haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -694,17 +710,25 @@ declsFromHsBinds ctx = dedupByTick . concatMap go
 isInstanceMethodVar :: Var -> Bool
 isInstanceMethodVar v = specialVar v == Just TypeclassySV
 
+-- | These are special bindings related to pattern synonyms which are not
+-- needed for us and contain overlapping declarations.
+isPatternSynonymSpecialVar :: Var -> Bool
+isPatternSynonymSpecialVar v = specialVar v == Just PatternSynonymSV
+
 specialVar :: Var -> Maybe SpecialVar
 specialVar v = case take 2 (nameOccurenceString (varName v)) of
     -- List appended as cases are encountered.
     "$d" -> Just $! TypeclassySV
     "$f" -> Just $! TypeclassySV
     "$c" -> Just $! TypeclassySV
+    "$m" -> Just $! PatternSynonymSV
```

Comment:
Could you include a link to where these magic $values are described into a comment?

Reply:
See comment "Making system names" in GHC source compiler/basicTypes/OccName.hs .

Reply:
I found related functions and updated the comment. See functions around https://hackage.haskell.org/package/ghc-8.6.5/docs/src/OccName.html#mkDataConWrapperOcc.